src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -226,6 +226,10 @@ ImmutableArray<ExpressionSyntax> assignments
 				immutable = false;
 			}
 
+			if( m_context.IsTotallyImmutableType( type ) ) {

```

Comment:
Should be something more like this:

```suggestion
			if( m_context.IsImmutable( type, ImmutableTypeKind.Total, () => nameSyntax.GetLocation(), ... ) ) {
```